presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java

```diff
@@ -217,6 +224,32 @@ public void close()
         stop();
     }
 
+    private boolean allContainersHealthy()
+    {
+        List<Container> containers = ImmutableList.copyOf(getContainers());
+        return containers.stream().allMatch(Environment::containerIsHealthy);
+    }
+
+    private static boolean containerIsHealthy(Container container)
+    {
+        try {
+            if (!container.isHealthy()) {
+                log.warn("Container %s is not healthy", container.getContainerName());
+                return false;
+            }
+        }
+        catch (RuntimeException ignored) {
```

Comment:
I don't like exception in non-exceptional flow of code. Can you just make container.isHealthy() return `true` when no healthcheck is defined. Or alternatively make `isHealthy` return `Optional<Boolean>` if you feel it is beneficial to know when we are in `unknown` state.